### PR TITLE
Fix flaky test related to tracking events

### LINKED ISSUE #3561: Fix flaky tracking integration tests
Postgres integration test suite 033 matches expected and actual tracking events. However, the experimental parser fires its tracking event non-deterministically. Commits 7a32129, and c43c5fa were supposed to fix this, yet [circleci](https://app.circleci.com/pipelines/github/dbt-labs/dbt/3023/workflows/cee21f06-bbc4-4b47-a12b-339259f430b9/jobs/44132) still sees inconsistent failures. 

This set of tests may not actually be all that useful as is and we may want to rework how we test events going forward anyway.

### PR DESCRIPTION
**Pull request by kwigley**
resolves #3561

### Description
- ~skips~ comment out tests related to event tracking, these are testing non-deterministic behavior
- allow null users when calling tracking event code, skip sending events downstream if this is the case

### Checklist
 - [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
 - [x] I have run this code in development and it appears to resolve the stated issue
 - [x] This PR includes tests, or tests are not required/relevant for this PR
 - [ ] I have updated the `CHANGELOG.md` and added information about my change to the "dbt next" section.

### CODE REVIEW

File: test/integration/033_event_tracking_test/test_events.py [hunk 1]
```diff
@@ -9,6 +10,7 @@
 import dbt.tracking
 import dbt.utils
 
+pytestmark = pytest.mark.skip("Experimental parser generates events non-deterministically, skipping.")
```
Comment (leahwicz) [inline test/integration/033_event_tracking_test/test_events.py:13]: Does this skip all the tests in this file? I don't see `pytestmark` being used anywhere so I'm not sure how that works

Comment (kwigley) [reply test/integration/033_event_tracking_test/test_events.py:13]: That is correct, it marks all tests in the module as skipped!

https://docs.pytest.org/en/stable/reference.html#globalvar-pytestmark

**Review by leahwicz — APPROVED** [review]

File: test/integration/033_event_tracking_test/test_events.py (hunk 1 above)
Comment (kwigley) [reply test/integration/033_event_tracking_test/test_events.py:13]: Well, I thought it did. https://github.com/dbt-labs/dbt/pull/3604/checks?check_run_id=3124252846

**Review by nathaniel-may — APPROVED** [review]

**Review by leahwicz — APPROVED** [review]

**Review by nathaniel-may — APPROVED** [review]
The fmt changes seem unintentional, but it doesn't actually matter. lgtm.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -6,28 +6,29 @@ name: dbt Tests
...
-    branches: 
-      - 'develop' 
-      - '*.latest' 
-      - 'pr/*'
-      - 'releases/*'
+    branches:
+      - "develop"
+      - "*.latest"
+      - "releases/*"
+  pull_request:
```
Comment (kwigley) [inline .github/workflows/tests.yml:13]: @leahwicz wanted to run this change by you. I was confused why tests weren't picking up my change, I realized the behavior for the `pull_request_target` event checks out the target branch for the checkout step. I think this is what we want, I'd like to confirm though!

**Comment by kwigley** [comment]
Created #3631 to address root cause.

**Pull request merged**